src/shared/vlan-util.c

```diff
@@ -61,6 +62,10 @@ int config_parse_vlanid(
                 return 0;
         }
         if (r < 0) {
+                if (streq(lvalue, "DefaultPVID") && streq(rvalue, "none")) {
+                        *id = -1;
```

Comment:
id is of type `*uint16_t`. It's not OK to assign -1 to it (didn't your compiler complain about this?)

Please define a new macro VLANID_DEFAULT or so, mapping to `((uint16_t) -1)` or so and use that at the reevant places.